### PR TITLE
docs: add Toy Tutorial Chapter 2

### PR DESCRIPTION
**Pull request by superlopuh**
Here is the long-awaited sequel to Toy Chapter 1, featuring Operation definition. A few of the APIs described in this notebook are under flux, but it still seems useful to add the rest of the Toy notebooks in and keep them updated as we go.

### CODE REVIEW

File: docs/Toy/Toy_Ch2.ipynb [hunk 1]
```diff
@@ -0,0 +1,452 @@
+{
```
Comment (webmiche) [inline docs/Toy/Toy_Ch2.ipynb:1]: <p>The formatting in this section is a bit weird. Is this intended or just a fragment of your import procedure?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/xdslproject/xdsl/pull/739/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (superlopuh) [reply docs/Toy/Toy_Ch2.ipynb:1]: <p>What do you mean by formatting? It's mostly text from the MLIR tutorial</p><p> </p>

Comment (webmiche) [reply docs/Toy/Toy_Ch2.ipynb:1]: There are some linebreaks that look weird to me.

Comment (superlopuh) [reply docs/Toy/Toy_Ch2.ipynb:1]: Can you give some examples? I don't see anything weird

Comment (webmiche) [reply docs/Toy/Toy_Ch2.ipynb:1]: Same. I guess it was a ReviewNB problem?  Anyway, you can ignore this comment :)

**Review by webmiche — APPROVED** [review]

**Comment by superlopuh** [comment]
Next one is about lowerings, which people were asking about, hopefully will be ready to put on the website soon

**Pull request merged**